From b19530b9b6fe42c8f6e4ffe538bbd53e87da0bc8 Mon Sep 17 00:00:00 2001
From: "Richard J. Knight" <rjknight@us.ibm.com>
Date: Wed, 1 Apr 2020 15:35:58 +0000
Subject: [PATCH] PyTorch TensorRT fix for GLIB_2.14 memcpy

- Use shim lib from our TensorRT package for the missing GLIBC_2.14
  memcpy symbol
---
 caffe2/CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/caffe2/CMakeLists.txt b/caffe2/CMakeLists.txt
index 535fe4f..6c4d78a 100644
--- a/caffe2/CMakeLists.txt
+++ b/caffe2/CMakeLists.txt
@@ -896,6 +896,15 @@ if(USE_OPENMP AND OPENMP_FOUND)
   target_link_libraries(torch_cpu PRIVATE ${OpenMP_CXX_LIBRARIES})
 endif()
 
+if (USE_TENSORRT)
+  find_library(MEMCPY_FIX_LIB libmemcpy-2.14.so)
+  target_link_libraries(torch_cuda PRIVATE ${MEMCPY_FIX_LIB})
+  message(STATUS "Added shared lib fix for memcpy@GLIBC_2.14" ${MEMCPY_FIX_LIB})
+else()
+  # Unneeded and not present if building without NVIDIA-built TensorRT
+  set(MEMCPY_FIX_LIB "")
+endif()
+
 
 if(USE_ROCM)
   target_compile_definitions(torch_hip PRIVATE
-- 
1.8.3.1

